### PR TITLE
[cairo] update toolchain

### PR DESCRIPTION
**Pull request by planetmarshall**
Specify library name and version:  **cairo**

Update the cairo meson build to the new toolchains

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by planetmarshall** [comment]
Failing due to https://github.com/conan-io/conan/issues/11867. Freetype needs updating.

**Comment by planetmarshall** [comment]
retry when #13355 is merged.

File: recipes/cairo/meson/conanfile.py [hunk 1]
```diff
@@ -138,100 +142,93 @@ def validate(self):
...
+        options["png"] = is_enabled(self.options.with_png)
+        options["freetype"] = is_enabled(self.options.with_freetype)
+        options["fontconfig"] = is_enabled(self.options.with_fontconfig)
         if self.settings.os == "Linux":
-            defs["xcb"] = boolean(self.options.get_safe("with_xcb"))
-            defs["xlib"] = boolean(self.options.get_safe("with_xlib"))
-            defs["xlib-xrender"] = boolean(self.options.get_safe("with_xlib_xrender"))
+            options["xcb"] = is_enabled(self.options.get_safe("with_xcb"))
+            options["xlib"] = is_enabled(self.options.get_safe("with_xlib"))
+            options["xlib-xrender"] = is_enabled(self.options.get_safe("with_xlib_xrender"))
```
Comment (prince-chrismc) [inline recipes/cairo/meson/conanfile.py:161]: can we lose get save since these should be present?

File: recipes/cairo/meson/conanfile.py [hunk 2]
```diff
@@ -138,100 +142,93 @@ def validate(self):
...
+        options["openvg"] = "disabled"  # https://www.khronos.org/openvg/
+        options["qt"] = "disabled"  # not yet compilable in cairo 1.17.4
+        options["gtk2-utils"] = "disabled"
+        options["spectre"] = "disabled"  # https://www.freedesktop.org/wiki/Software/libspectre/
+
+        meson = MesonToolchain(self)
+        meson.project_options.update(options)
+
+        if is_apple_os(self) and Version(self.version) <= "1.17.4":
+            # This was fixed in the meson build from 1.17.6
```
Comment (prince-chrismc) [inline recipes/cairo/meson/conanfile.py:192]: does not exactly match comment?

```suggestion
        if is_apple_os(self) and Version(self.version) < "1.17.6":
            # This was fixed in the meson build from 1.17.6
```

File: recipes/cairo/meson/test_v1_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1.2)
+project(test_package C)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(cairo CONFIG REQUIRED)
+
+add_executable(${PROJECT_NAME} ../test_package/test_package.c)
```
Comment (prince-chrismc) [inline recipes/cairo/meson/test_v1_package/CMakeLists.txt:9]: Just FYI, there as a new way for this https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/test_v1_package/CMakeLists.txt

less code :)

**Review by prince-chrismc — COMMENTED** [review]
One question https://github.com/conan-io/conan-center-index/pull/13427/files#r1004991794

Otherwise it's perfect

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**